src/model/index.js

```diff
@@ -334,8 +334,12 @@ export class ZOAuthModel {
     if (accessToken) {
       await sessions.nextItem((a) => {
         if (a.access_token === accessToken) {
-          access = a;
-          return true;
+          const aDate = new Date(a.created);
+          aDate.setSeconds(aDate.getSeconds() + a.expires_in);
```

Comment:
So why are you using `Date` here? You get unix time value in `a.created`, so you could simply do an addition (convert `expires_in` in milliseconds first then add it to `created`), and compare the result to `Date.now()`.

Reply:
I transform `TimeStamp` into `Date` for be able to use `.setSeconds()` and `.getSeconds()` methods.

But you're right, I can do that to use only TimeStamp:
```
const aDate = a.created + (a.expires_in*1000);
if(aDate > new Date().getTime()) {
    access = a;
    return true;
}
```

Reply:
> But if the database don't return TimeStamp it failed.

this should not happen.

Reply:
and I think `new Date().getTime()` == `Date.now()`

Reply:
The database stores the "created" attribute in `DateTime` format, which normally returns the date in `ISO` format , but we have seen that the database can return `TimeStamp` in some case. So I decided to turn all the variables in `Date` to avoid bugs.

Reply:
It has since been fixed in zoapp-core `0.7.0`.

Reply:
So you're supposed to change this code to only deal with unix time since that's what zoapp-core returns. The database stores the dates correctly, and it also returns them correctly. It is zoapp-core that converts them to unix time values and we should now assume that these values will always be unix time values.